modules/mauth-authenticator-http4s/src/main/scala/com/mdsol/mauth/http4s/MauthPublicKeyProvider.scala

```diff
@@ -0,0 +1,80 @@
+package com.mdsol.mauth.http4s
+
+import cats.ApplicativeThrow
+import cats.effect.Async
+import com.mdsol.mauth.http4s.client.Implicits.NewSignedRequestOps
+import com.mdsol.mauth.models.UnsignedRequest
+import com.mdsol.mauth.scaladsl.utils.ClientPublicKeyProvider
+import com.mdsol.mauth.util.MAuthKeysHelper
+import com.mdsol.mauth.{AuthenticatorConfiguration, MAuthRequestSigner}
+import org.http4s.client.Client
+import org.http4s.{EntityDecoder, Response, Status}
+import scalacache.memoization.memoizeF
+import scalacache.Cache
+
+import java.net.URI
+import java.security.PublicKey
+import java.util.UUID
+import scala.concurrent.duration._
+import cats.implicits._
+import com.mdsol.mauth.http4s.MauthPublicKeyProvider.SecurityToken
+import io.circe.{Decoder, HCursor}
+import org.http4s.circe.jsonOf
+import org.typelevel.log4cats.Logger
+
+class MauthPublicKeyProvider[F[_]: Async: Logger](configuration: AuthenticatorConfiguration, signer: MAuthRequestSigner, val client: Client[F])(implicit
```

Comment:
this can be then 
```suggestion
class MauthPublicKeyProvider[F[_]: MonadThrow: Logger](configuration: AuthenticatorConfiguration, signer: MAuthRequestSigner, val client: Client[F])(implicit
```